### PR TITLE
improve fzf 

### PR DESCRIPTION
**Pull request by pittcat**

### CODE REVIEW

File: layers/+tools/fzf/config.vim [hunk 1]
```diff
@@ -23,5 +23,56 @@ else
...
+    \ 'source':  'ls -1 ~/.vim/session',
+    \ 'sink':    'SLoad',
+    \ 'options': '+m --prompt="Sessions> "',
+    \ 'down':    '40%'
+    \})
+  endfunction
+  command! Sessions call s:sessions()
+  nnoremap <silent> <leader>sf :Sessions<cr>
+
+  let g:fzf_colors =
```
Comment (liuchengxu) [inline layers/+tools/fzf/config.vim]: `g:fzf_colors` has been already configured, see https://github.com/liuchengxu/space-vim/blob/3bbf9f50de9b37b8a5ca8bcabf11c26faa08fa88/layers/%2Btools/fzf/config.vim#L13.

File: layers/+tools/fzf/config.vim [hunk 2]
```diff
@@ -23,5 +23,56 @@ else
...
+  function! s:sessions()
+    call fzf#run({
+    \ 'source':  'ls -1 ~/.vim/session',
+    \ 'sink':    'SLoad',
+    \ 'options': '+m --prompt="Sessions> "',
+    \ 'down':    '40%'
+    \})
+  endfunction
+  command! Sessions call s:sessions()
+  nnoremap <silent> <leader>sf :Sessions<cr>
```
Comment (liuchengxu) [inline layers/+tools/fzf/config.vim]: Move this block, `s:sessions()` to `core/autoload/spacevim/plug/fzf.vim`, `spacevim#plug#fzf#Session()`.

```vim
nnoremap <silent> <Leader>fs :call spacevim#plug#fzf#Session()<CR>
```

**Comment by liuchengxu** [comment]
Please split this single PR into 2. One PR for tmux specific and the other for fzf related.

**Comment by pittcat** [comment]
it seems like that "<leader>fs "  has been mapped.
[better-defaults](https://github.com/liuchengxu/space-vim/blob/master/layers/%2Bdistributions/better-defaults/README.md)

File: core/autoload/spacevim/plug/fzf.vim [hunk 3]
```diff
@@ -190,6 +190,34 @@ function! s:align_lists(lists)
...
+  \   <bang>0 ? fzf#vim#with_preview('up:70%')
+  \           : fzf#vim#with_preview('right:50%:hidden', '?'),
+  \   <bang>0)
+
+" Likewise, Files command with preview window
+command! -bang -nargs=? -complete=dir Files
+  \ call fzf#vim#files(<q-args>, fzf#vim#with_preview(), <bang>0)
+" }
+
+function! s:spacevim#plug#fzf#Session()
```
Comment (liuchengxu) [inline core/autoload/spacevim/plug/fzf.vim]: Remove `s:`, `s:spacevim#plug#fzf#Session()` -> `spacevim#plug#fzf#Session()`.

Comment (pittcat) [reply core/autoload/spacevim/plug/fzf.vim]: sorry

File: core/autoload/spacevim/plug/fzf.vim [hunk 4]
```diff
@@ -190,6 +190,34 @@ function! s:align_lists(lists)
   return a:lists
 endfunction
 
+autocmd VimEnter * command! -bang -nargs=* Ag
```
Comment (liuchengxu) [inline core/autoload/spacevim/plug/fzf.vim]: `autocmd VimEnter *` is useless in the `autoload` mechanism. So, remove `autocmd VimEnter *`.

**Comment by pittcat** [comment]
To be honest,use the custom fzf colors maybe more reasonable，as follow,
" Customize fzf colors to match your color scheme
let g:fzf_colors =
\ { 'fg':      ['fg', 'Normal'],
  \ 'bg':      ['bg', 'Normal'],
  \ 'hl':      ['fg', 'Comment'],
  \ 'fg+':     ['fg', 'CursorLine', 'CursorColumn', 'Normal'],
  \ 'bg+':     ['bg', 'CursorLine', 'CursorColumn'],
  \ 'hl+':     ['fg', 'Statement'],
  \ 'info':    ['fg', 'PreProc'],
  \ 'border':  ['fg', 'Ignore'],
  \ 'prompt':  ['fg', 'Conditional'],
  \ 'pointer': ['fg', 'Exception'],
  \ 'marker':  ['fg', 'Keyword'],
  \ 'spinner': ['fg', 'Label'],
  \ 'header':  ['fg', 'Comment'] 
，because if you set other colortheme ,the object which is matched by fzf looks like vague.

**Comment by liuchengxu** [comment]
Yeah, space-vim only guarantee the default `g:fzf_colors` is suitable for space-vim-dark. User can override `g:fzf_colors` in `UserConfig` if it looks strange.

**Comment by pittcat** [comment]
Ok，it's no uncontroversial any more.

**Comment by liuchengxu** [comment]
Merged, thanks!

**Pull request merged**